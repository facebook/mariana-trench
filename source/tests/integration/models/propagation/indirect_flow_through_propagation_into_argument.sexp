(method (private) "LFlow;.flow:()V"
 (
  LINE(1)
  (invoke-virtual (v0) "LSource;.source:()LData;")
  (move-result-object v0)

  LINE(2)
  (invoke-virtual (v0) "LData;.propagation_this:()LData;")
  (move-result-object v3)

  LINE(3)
  (invoke-virtual (v2 v3) "LSink;.sink:(LData;)V")
  (return-void)
 )
)
