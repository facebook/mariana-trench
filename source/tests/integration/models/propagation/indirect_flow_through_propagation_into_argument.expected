{
  "metadata" :
  {
    "@generated" : true
  },
  "models" :
  [
    {
      "issues" :
      [
        {
          "position" :
          {
            "line" : 3,
            "path" : "Flow.java"
          },
          "rule" : 1,
          "sinks" :
          [
            {
              "call_position" :
              {
                "line" : 3,
                "path" : "Flow.java"
              },
              "callee" : "LSink;.sink:(LData;)V",
              "callee_port" : "Argument(1)",
              "distance" : 1,
              "kind" : "TestSink",
              "origins" :
              [
                "LSink;.sink:(LData;)V"
              ]
            }
          ],
          "sources" :
          [
            {
              "call_position" :
              {
                "line" : 1,
                "path" : "Flow.java"
              },
              "callee" : "LSource;.source:()LData;",
              "callee_port" : "Return",
              "distance" : 1,
              "kind" : "TestSource",
              "local_positions" :
              [
                {
                  "line" : 2
                }
              ],
              "origins" :
              [
                "LSource;.source:()LData;"
              ]
            }
          ]
        }
      ],
      "method" : "LFlow;.flow:()V",
      "position" :
      {
        "line" : 0,
        "path" : "Flow.java"
      }
    }
  ]
}
